src/transaction_builder.js

```diff
@@ -686,7 +682,13 @@ TransactionBuilder.prototype.sign = function (vin, keyPair, redeemScript, hashTy
 
   var kpPubKey = keyPair.getPublicKeyBuffer()
   if (!canSign(input)) {
-    prepareInput(input, kpPubKey, redeemScript, witnessValue, witnessScript)
+    if (witnessValue !== undefined) {
+      if (input.value !== undefined && input.value !== witnessValue) throw new Error('Input didn\'t match witnessValue')
+      typeforce(types.Satoshi, witnessValue)
+      input.value = witnessValue
+    }
```

Comment:
@afk11

Reply:
I think this side steps the call to fixMultisigOrder, which would remove any signatures from the array which aren't valid?

Reply:
@afk11 this isn't related to MS order,  it simply ensures that `.value` is added on if missing